src/main/groovy/com/pstehlik/groovy/gelf4j/appender/GelfMessageBuilder.groovy

```diff
@@ -0,0 +1,103 @@
+package com.pstehlik.groovy.gelf4j.appender
+
+import org.apache.log4j.Layout
+import org.apache.log4j.spi.LoggingEvent
+import org.apache.log4j.spi.ThrowableInformation
+
+class GelfMessageBuilder {
+
+  private static final String GELF_VERSION = '1.1'
+
+  public static final Integer SHORT_MESSAGE_LENGTH = 250
+
+  Map<String, String> message
+
+  GelfMessageBuilder(Gelf4JAppender appender, LoggingEvent loggingEvent) {
+
+    message = [:]
+    setField('level', loggingEvent.level.syslogEquivalent as String)
+    setField('timestamp', new BigDecimal(Math.floor(loggingEvent.getTimeStamp() / 1000)) as String)
+    setField('version', GELF_VERSION)
+
+    if (appender.includeLocationInformation && loggingEvent.locationInformationExists()) {
+      setAdditionalField('code_location_file', loggingEvent.locationInformation.fileName)
+      setAdditionalField('code_location_line', loggingEvent.locationInformation.lineNumber)
+      setAdditionalField('code_location_method_name', loggingEvent.locationInformation.methodName)
+    }
+
+    setAdditionalField('logger_name', loggingEvent.loggerName)
+    setAdditionalField('thread_name', loggingEvent.threadName)
```

Comment:
Added additional information which may prove helpful in debugging based on log messages